### PR TITLE
812: Fix profile display when no user is selected...

### LINKED ISSUE #812: GP_Route_Profile::profile_view() doesn't check if user exists
`wp_get_current_user()` returns an empty `WP_User` object which skips the `! $user` condition. We need an `$user->exists()` check here.

### PR DESCRIPTION
**Pull request by toolstack**
and no one is logged in.

Resolves #812.

### CODE REVIEW

**Comment by toolstack** [comment]
Note the TravisCI failures are for deprecated functions in 7.2, I've opened issue #941 to take care of them.

File: gp-includes/routes/profile.php [hunk 1]
```diff
@@ -40,7 +40,7 @@ public function profile_view( $user = '' ) {
 			$user = get_user_by( 'slug', $user );
 		}
 
-		if ( ! $user ) {
+		if ( ! $user || ( is_object( $user ) && ! $user->exists() ) ) {
```
Comment (ocean90) [inline gp-includes/routes/profile.php]: Should check explicit for `WP_User`: `$user instanceof WP_User`

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: gp-includes/routes/profile.php (hunk 1 above)
Comment (toolstack) [reply gp-includes/routes/profile.php]: Added.

**Comment by Mte90** [comment]
I am wondering if there are part of glotpress where this kind of check can be useful. I will investigate asap about it.

**Review by Mte90 — APPROVED** [review]
For me is perfect!
Checked in the code and we don't have other cases where we need to ensure that the user exist and is a valid object.

**Pull request merged**